### PR TITLE
Better GitHub username check

### PR DESCRIPTION
**Pull request by bjoernhaeuser**
Replaces #573, #575 

I forked the repository and added the tests, as described in #575.

### CODE REVIEW

**Comment by bjoernhaeuser** [comment]
@thomas-riccardi take a look here please :)

**Comment by thomas-riccardi** [comment]
Seems fine ! Thanks for writing the tests.

**Comment by samrocketman** [comment]
Let's squash this into two commits.  One for @thomas-riccardi and one for @bjoernhaeuser .  If you don't know how to squash, then I can do it.

**Comment by bjoernhaeuser** [comment]
@samrocketman done!

File: src/test/java/org/jenkinsci/plugins/ghprb/GhprbTriggerTest.java [hunk 1]
```diff
@@ -1,52 +1,60 @@
...
+import static org.mockito.Mockito.mock;
+
 /**
  * Unit test for {@link org.jenkinsci.plugins.ghprb.GhprbTriggerTest}.
  */
 @RunWith(MockitoJUnitRunner.class)
-public class GhprbTriggerTest {
-
+public class GhprbTriggerTest
+{
```
Comment (samrocketman) [inline src/test/java/org/jenkinsci/plugins/ghprb/GhprbTriggerTest.java]: Best to keep the style consistent with the rest of the code base.  Looking at other files the open bracket for classes is on the same line as the class definition.

I'll only comment this once but this same feedback applies to other parts of the code in this change.

**Comment by bjoernhaeuser** [comment]
I will wait for a decision on #599 and afterwards continue here.

**Comment by bjoernhaeuser** [comment]
I rebased to current master, the style changes are now incorporated.

**Comment by thomas-riccardi** [comment]
Thanks @bjoernhaeuser for properly finishing what I started !

**Pull request merged**